src/old/modules/materials/components/MaterialsView.tsx

```diff
@@ -78,58 +80,51 @@ const MaterialsView: React.FC = () => {
   const classes = useStyles();
   const materials = selectPostsByIds(postIds);
   const origins = useSelector(selectOrigins);
-  const originsInPlural: IOrigin[] = [];
+  let originsInPlural: IOrigin[] = [];
 
   if (origins.length) {
-    const el1: IOrigin = { ...origins[0] };
-    const el2: IOrigin = { ...origins[1] };
-    const el3: IOrigin = { ...origins[2] };
-
-    Object.defineProperty(el1, 'name', {
-      enumerable: false,
-      configurable: true,
-      writable: true,
-      value: `${t(langTokens.experts.expertOpinion, defaultPlural)}`,
+    const translatedOrigins = origins.map((origin) => {
+      const translations = {
+        '1': `${t(langTokens.experts.expertOpinion, defaultPlural)}`,
+        '3': `${t(langTokens.common.translation, defaultPlural)}`,
+      };
+      const translatedOrigin = { ...origin };
+      translatedOrigin.name = translations[origin.id] || origin.name;
+      return translatedOrigin;
```

Comment:
This block of code is completely unreadable. Using `let` and do a conditional assignment is an antipattern. 

I suggest at least leaving a comment describing what this does and leaving a `@TODO` comment for future refactoring.